starter-code/src/clue.js

```diff
@@ -1,94 +1,175 @@
+  { name: "candlestick", weight: 2 },
+  { name: "dumbbell", weight: 30 },
+  { name: "poison", weight: 2 },
+  { name: "axe", weight: 15 },
+  { name: "bat", weight: 13 },
+  { name: "trophy", weight: 25 },
+  { name: "pistol", weight: 20 }
+];
 
-// Weapons
+//--------------------//
+// Random selector
+//--------------------//
+function randomSelector(array) {
+  // random number 0 till length - 1 (floor). random number is 0 till 0.99
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+ 
+//--------------------//
+// Create a mystery (the object way)
+// This was version 1. As an object. Though Jasmine demanded an array :) see below.
+//--------------------//
+// function pickMistery() {
+//   var envelop = {};
+//   envelop.character = randomSelector(charactersArray);
+//   envelop.weapon = randomSelector(weaponsArray);
+//   envelop.room = randomSelector(roomsArray);
+//   return envelop;
+// };
 
-// Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+// var misteryEnvelop = pickMistery();
+// console.log(misteryEnvelop);
 
-// Characters Collection
-var charactersArray = [];
+// // Reveal the mystery
+// console.log(misteryEnvelop.character.first_name + " " + misteryEnvelop.character.last_name +
+//   " killed Mr.Boddy using the " + misteryEnvelop.weapon.name + " in the " + misteryEnvelop.room.name + "!!!!");
 
-// Rooms' Collection
-var roomsArray = [];
 
-// Weapons Collection
-var weaponsArray = [];
+//--------------------//
+// Create a mystery (the array way)
+//--------------------//
+function pickMistery() {
+  var envelop = [];
+  envelop.push(randomSelector(charactersArray));
+  envelop.push(randomSelector(weaponsArray));
+  envelop.push(randomSelector(roomsArray));
+  return envelop;
+};
+
+var misteryEnvelop = pickMistery();
+console.log(misteryEnvelop);
+
+//--------------------//
+// Reveal the mystery
+//--------------------//
+function revealMistery(misteryEnvelop) {
+  return (misteryEnvelop[0].first_name + " " + misteryEnvelop[0].last_name +
+    " killed Mr.Boddy using the " + misteryEnvelop[1].name + " in the " + misteryEnvelop[2].name + "!!!!");
+};
+
+console.log(revealMistery(misteryEnvelop));
+
+ 
+ 
+//--------------------//
+// js to HTML
+//--------------------//
+// Character
+document.getElementById("character").src = misteryEnvelop[0].image;
+// document.getElementById("character-name").innerHTML = (misteryEnvelop[0].first_name + " " + misteryEnvelop[0].last_name); 
```

Comment:
Try not to leave comments in code that is committed.   If you want to to keep code that you are not using, try using `git stash` . This command allows you to "stash" all uncommitted changes.  The changes can then be reinstated using `git stash pop` (which reinstates and also deletes the last stash).  Maybe read a little about it on [free code camp](https://medium.freecodecamp.org/useful-tricks-you-might-not-know-about-git-stash-e8a9490f0a1a) :)